lib/rule-executors.ts

```diff
@@ -111,3 +106,56 @@ export function undoPrehide(): boolean {
   }
   return !!existingElement;
 }
+
+function querySingleReplySelector(selector: string, parent: ParentNode = document): HTMLElement[] {
+  if (selector.startsWith('aria/')) {
+    return []
+  }
+  if (selector.startsWith('xpath/')) {
+    const xpath = selector.slice(6)
+    const result = document.evaluate(xpath, parent, null, XPathResult.ANY_TYPE, null)
+    let node: Node = null
+    const elements: HTMLElement[] = []
+    // eslint-disable-next-line no-cond-assign
+    while (node = result.iterateNext()) {
+      elements.push(node as HTMLElement)
+    }
+    return elements
+  }
+  if (selector.startsWith('text/')) {
+    return []
+  }
+  if (selector.startsWith('pierce/')) {
+    return []
+  }
+  return Array.from(parent.querySelectorAll(selector))
+}
+
+function querySelectorChain(selectors: string[]): HTMLElement[] {
+  let parent: ParentNode = document
+  let matches: HTMLElement[]
+  for (const selector of selectors) {
+    matches = querySingleReplySelector(selector, parent)
+    if (matches.length === 0) {
+      return []
+    }
+    parent = matches[0]
+  }
+  return matches;
+}
+
+export function elementSelector(selector: ElementSelector): HTMLElement[] {
+  if (typeof selector === 'string') {
+    return Array.from(document.querySelectorAll(selector))
+  }
+  // selector is an array of @puppeteer/replay.Selector
+  for (const replySelector of selector) {
```

Comment:
So when selector is an array of `replay.Selector`, we assume that all elements are equivalent and use the first one that works?

Reply:
Yes. The algorithm is described in this comment: https://github.com/puppeteer/replay/blob/main/src/Schema.ts#L82-L94